mm2src/coins/eth.rs

```diff
@@ -1457,21 +1458,19 @@ impl WatcherOps for EthCoin {
                 )));
             }
 
-            if tx_from_rpc.to != Some(expected_swap_contract_address) {
-                if tx_from_rpc.to == fallback_swap_contract {
-                    expected_swap_contract_address = fallback_swap_contract.ok_or_else(|| {
-                        ValidatePaymentError::InternalError("Fallback swap contract address not found".to_string())
-                    })?;
-                } else {
-                    return MmError::err(ValidatePaymentError::WrongPaymentTx(format!(
-                        "Payment tx {:?} was sent to wrong address, expected {:?}",
-                        tx_from_rpc, expected_swap_contract_address,
-                    )));
-                }
+            if tx_from_rpc.to != Some(expected_swap_contract_address) && tx_from_rpc.to != fallback_swap_contract {
+                return MmError::err(ValidatePaymentError::WrongPaymentTx(format!(
+                                      "Payment tx {tx_from_rpc:?} was sent to wrong address, expected either {expected_swap_contract_address:?} or the fallback {fallback_swap_contract:?}"
```

Comment:
I guess you copied my suggestion as it is.
Could you please fix the tabulation?

Reply:
Yeah I copied your suggestion and made changes on it. Then I falsely assumed that cargo fmt would fix everything.